bridges/snowbridge/pallets/system-v2/src/lib.rs

```diff
@@ -141,40 +139,36 @@ pub mod pallet {
 		/// Sends a command to the Gateway contract to instantiate a new agent contract representing
 		/// `origin`.
 		///
-		/// Fee required: Yes
-		///
-		/// - `origin`: Must be `Location` of a sibling parachain
+		/// - `location`: The location representing the agent
+		/// - `fee`: Ether to pay for the execution cost on Ethereum
 		#[pallet::call_index(1)]
 		#[pallet::weight(T::WeightInfo::create_agent())]
 		pub fn create_agent(
 			origin: OriginFor<T>,
 			location: Box<VersionedLocation>,
 			fee: u128,
 		) -> DispatchResult {
-			let _ = match origin.clone().into() {
-				Ok(RawOrigin::Root) => Ok(Here.into()),
-				_ => T::SiblingOrigin::ensure_origin(origin),
-			}?;
+			T::SiblingOrigin::ensure_origin(origin)?;
```

Comment:
ditto, see comment above:

```suggestion
			T::FrontendOrigin::ensure_origin(origin)?;
```